File: DimoTalk.Maui/Models/Conversation.cs
namespace DimoTalk.Maui.Models;

public class Conversation
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string UserId { get; set; } = string.Empty;
    public string Title { get; set; } = "新对话";
    public List<Message> Messages { get; set; } = new();
    public DateTime CreatedAt { get; set; } = DateTime.Now;
    public DateTime UpdatedAt { get; set; } = DateTime.Now;

    public void AddMessage(Message msg)
    {
        Messages.Add(msg);
        UpdatedAt = DateTime.Now;
    }
}

File: Hardwaver/lib/models/conversation.dart
import 'package:uuid/uuid.dart';
import 'message.dart';

class Conversation {
  final String id;
  final String userId;
  final String title;
  final List<Message> messages;
  final DateTime createdAt;
  DateTime updatedAt;

  Conversation({
    String? id,
    required this.userId,
    this.title = '新对话',
    List<Message>? messages,
    DateTime? createdAt,
    DateTime? updatedAt,
  })  : id = id ?? const Uuid().v4(),
        messages = messages ?? [],
        createdAt = createdAt ?? DateTime.now(),
        updatedAt = updatedAt ?? DateTime.now();

  void addMessage(Message msg) {
    messages.add(msg);
    updatedAt = DateTime.now();
  }
}
